### PR TITLE
DisplayNickname Fix

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

**Comment by louis1706** [comment]
tested by 1312255201 in https://discord.com/channels/656673194693885975/1073392577245610035/1073916649591296080

**Review by xNexus-ACS — APPROVED** [review]

**Comment by louis1706** [comment]
Tested on my server it's always show * even if player don't have a custom name

**Comment by louis1706** [comment]
I will send fix tomorrow

**Comment by louis1706** [comment]
I will say now would work perfectly fine just different than before

File: Exiled.CustomItems/Events/PlayerHandler.cs [hunk 1]
```diff
@@ -19,36 +19,26 @@ namespace Exiled.CustomItems.Events
...
     internal sealed class PlayerHandler
     {
-        /// <inheritdoc cref="ChangingRoleEventArgs"/>
-        public void OnChangingRole(ChangingRoleEventArgs ev)
+        /// <inheritdoc cref="ChangingItemEventArgs"/>
+        public void OnChangingItem(ChangingItemEventArgs ev)
         {
-            if (ev.NewRole == RoleTypeId.Spectator)
+            if (ev.IsAllowed)
+                return;
```
Comment (JasKill) [inline Exiled.CustomItems/Events/PlayerHandler.cs:26]: Is it supposed to be like this?

Comment (louis1706) [reply Exiled.CustomItems/Events/PlayerHandler.cs:26]: i mean if we got an ChangedItemEventArgs it's will not need this

**Comment by louis1706** [comment]
i have tested on my server i have fixed the error getting by Null item 
now should work fine (it's would be nice if someone can test it with CustomItem i don't have any)

**Review by IRacle1 — APPROVED** [review]

**Comment by louis1706** [comment]
> Needs tested with custom items before being merged.

Tested with Nexus https://cdn.discordapp.com/attachments/990672706762575872/1076931642104500355/image.png

File: Exiled.CustomItems/Events/PlayerHandler.cs [hunk 2]
```diff
@@ -19,36 +19,26 @@ namespace Exiled.CustomItems.Events
...
+        }
 
-                    ev.Player.SendFakeSyncVar(player.ReferenceHub.networkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_displayName), player.DisplayNickname);
-                }
+        private void SpectatorCustomNickname(Player player, string itemName)
+        {
+            foreach (Player spetator in Player.List)
+            {
+                spetator.SendFakeSyncVar(player.NetworkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_displayName), itemName);
             }
```
Comment (iopietro) [inline Exiled.CustomItems/Events/PlayerHandler.cs:42]:
```suggestion
            foreach (Player spectator in Player.List)
                spectator.SendFakeSyncVar(player.NetworkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_displayName), itemName);
```

**Review by iopietro — DISMISSED** [review]

**Review by JasKill — APPROVED** [review]

**Review by Thundermaker300 — APPROVED** [review]

**Review by iopietro — APPROVED** [review]

**Pull request merged**